lib/audited/audit.rb

```diff
@@ -19,7 +19,12 @@ class YAMLIfTextColumnType
     class << self
       def load(obj)
         if text_column?
-          ActiveRecord::Coders::YAMLColumn.new(Object).load(obj)
+          begin
+            ActiveRecord::Coders::YAMLColumn.new(Object).load(obj)
+          rescue Psych::SyntaxError
+            obj = obj.gsub("=>", ":")
+            ActiveRecord::Coders::YAMLColumn.new(Object).load(obj)
+          end
```

Comment:
First, Can we do it this way? To avoid parsing it twice? 

```suggestion
          begin
            obj = obj.gsub("=>", ":")
            ActiveRecord::Coders::YAMLColumn.new(Object).load(obj)
          end
```

Second, what happens if we have a value like this `"{"a" => "some value include =>"}"? Could you test it?

Reply:
1. I don't want to modify a correct JSON parseable string which contains => for eq "{"a": "sample=>text"}"
2. I don't think "{"a" => "some value include =>"}" is possible as the incorrect data also has pattern which is hash.to_s

Reply:
> 1. I don't want to modify a correct JSON parseable string which contains => for eq "{"a": "sample=>text"}"
> 2. I don't think "{"a" => "some value include =>"}" is possible as the incorrect data also has pattern which is hash.to_s

why is "sample=>text" possible in the first point and not possible in the 2nd point? It should be possible in both, that's why we should find a safer way to replace "=>" with ":" in both cases. 

Consider the following: 

```ruby
"{"a": "value of a has a =>"}"
"{"a" => "value of a has a =>"}"
```

both should be parsed as

```ruby
"{"a": "value of a has a =>"}"
"{"a": "value of a has a =>"}"
```
but now, they are parsed as: 

```ruby
"{"a": "value of a has a =>"}"
"{"a": "value of a has a :"}" # this is wrong
```

> I don't want to modify a correct JSON

If a json is already correct, then the **proper** `replace` function should not modify it. If a json contains "=>" in its value, we should not replace it.

Reply:
I think a proper test coverage would be better